TempleEight/docs/getting-started.md

```diff
@@ -4,3 +4,400 @@ title: Getting Started
+
+#### Foreign Keys
+
+Also visible in the `Match` service is an example of a foreign key service property. A match is a relationship made between two `User`s, and their `id`s are stored in the respective fields. Using a foreign key will cause Temple to perform cross-service communication in order to verify the existence of any entities referred to, before this entity is created. For example, when the `/match` endpoint is invoked with a POST request for `User`s `1` and `2`, the Match service will first check with the `User` service that those `User`s actually exist, before creating the `Match`.
+
+#### Property Metadata
+
+The final thing we can see in the example Templefile is the `@serverSet` metadata on `Match` service's `matchedOn` field. Metadata beginning with `@` refer to a particular property, and tell us about this property's values and how they are accessed. 
+
+In this example, the `@serverSet` metadata is used. This means that the `matchedOn` value isn't expected to be provided in the POST request that creates this `Match` entity, and is instead calculated by the server itself when the request arrives. 
+
+## Running The Example
+
+Now that we have created our example Templefile, save it in a file named `dating-app.temple`, in our `temple-tutorial` directory created earlier.
+
+### Validating the Templefile
+
+The Temple CLI contains a tool for validating your Templefile before generating any code. We can invoke this tool by running:
+
+```
+~/Documents/temple-tutorial ❯❯❯ temple validate dating-app.temple
+Templefile validated correctly
+```
+
+Which means our Templefile is syntactically correct, and we an begin to generate all the code we need.
+
+### Generating Code
+
+At this point, we can begin to generate some code. Running: 
+
+```
+~/Documents/temple-tutorial ❯❯❯ temple generate dating-app.temple
+```
+
+Begins the process. The first thing we'll see is the CLI prompting us to give some further generation details that are specific to the metadata selections made in our Templefile. 
+
+```
+~/Documents/temple-tutorial ❯❯❯ temple generate dating-app.temple
+What should the Go module name be? (expected format "github.com/username/repo")
+```
+
+We need to enter the name of the Golang module we'll be generating into for our project, this can be anything, but standards dictate it should be the github URL of where this project will be hosted. 
+
+For this example enter `github.com/temple/tutorial`. 
+
+```
+~/Documents/temple-tutorial ❯❯❯ temple generate dating-app.temple
+What should the Go module name be? (expected format "github.com/username/repo")
+github.com/temple/tutorial
+github.com/temple/tutorial
+Generated project in /path/Documents/temple-tutorial
+```
+
+Now we can view the generated outputs of our project:
+
+```
+~/Documents/temple-tutorial ❯❯❯ tree
```

Comment:
This is going to confuse people if they don’t have `tree` installed, and don’t know what it is